### PR TITLE
Fix salt event queue (bsc#1203532)

### PR DESCRIPTION
**Pull request by nadvornik**
## What does this PR change?

This PR fixes the following problems with salt event queue:

1. Event ordering 

Events were ordered by minion ID. As a result, certain minions had low prority and their events were processed with extreme delays on loaded server.
This behavior was introduced by commit 6c7981570024 but it is no longer needed because 63c9aff6a9ef implements better fix.
This PR reverts the behavior to standard FIFO.

2. Detecting stalled events in `startConnectionWatchdog`

The methods `executor.getTaskCount()` and `executor.getCompletedTaskCount()` are unreliable - according documentation they return "approximate total number of tasks".  Customer's log shows that this is a real problem, it contains messages mentioning negative number of events: `Found [0, 0, 0, 0, 2, 0, -1, 0, 0] events without a job. Scheduling...`
This PR triggers the watchdog if there are queued events and the corresponding thread is idle.

3. Handling exceptions in `startConnectionWatchdog`

The watchdog thread can exit on unhandled exceptions. This PR attempts to catch all exceptions.

## GUI diff

No difference.

## Documentation
- No documentation needed: 

## Test coverage
- No tests: the issues are reproducible only on loaded SUMA server

- [ ] **DONE**

## Links

Partially fixes https://github.com/SUSE/spacewalk/issues/19063  (bsc#1203532)

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/suse/manager/reactor/PGEventStream.java [hunk 1]
```diff
@@ -123,17 +123,13 @@ public void run() {
...
                         // this can only happen in case we lost notifications somehow
                         List<Long> allJobs = SaltEventFactory.countSaltEvents(THREAD_POOL_SIZE + 1);
 
-                        List<Long> queuedJobs = executorServices.stream()
-                                .map(executor -> executor.getTaskCount() - executor.getCompletedTaskCount())
-                                .collect(Collectors.toList());
-
                         List<Long> missingJobs = IntStream.range(0, allJobs.size())
-                            .mapToObj(i -> allJobs.get(i) - queuedJobs.get(i))
+                            .mapToObj(i -> executorServices.get(i).getActiveCount() > 0 ? 0 : allJobs.get(i))
```
Comment (rjmateus) [inline java/code/src/com/suse/manager/reactor/PGEventStream.java:132]: @nadvornik could you explain the logic behind this if statement? Why if active count is bigger then 0 the missing jobs are 0? The all jobs should not be taken in consideration also?

**Review by rjmateus — COMMENTED** [review]
Just one question

File: java/code/src/com/suse/manager/reactor/PGEventStream.java (hunk 1 above)
Comment (nadvornik) [reply java/code/src/com/suse/manager/reactor/PGEventStream.java:132]: Active count is bigger then 0 means that the thread is running and processing events.
So if the thread is running, then everything is OK and no action is needed.
When the thread stopped and there are queued events - it means that something went wrong and the best we can do is to schedule all queued events.

**Review by rjmateus — APPROVED** [review]
LGTM

**Comment by cbosdo** [comment]
You just miss a changelog entry and it's all fine

**Comment by aaannz** [comment]
I guess this can be merged once change log entry is added. Problem with changing number of threads can be solved in different PR.

**Pull request merged**